pkg/discovery/worker/aggregation/container_usage_data_aggregator.go

```diff
@@ -62,13 +69,20 @@ func (maxUsageDataAggregator *maxUsageDataAggregator) String() string {
 }
 
 func (maxUsageDataAggregator *maxUsageDataAggregator) Aggregate(resourceMetrics *repository.ContainerMetrics) (float64, float64, float64, error) {
-	if len(resourceMetrics.Used) == 0 {
-		err := fmt.Errorf("error to aggregate container usage data using %s: used data points list is empty", maxUsageDataAggregator)
+	if len(resourceMetrics.Used) == 0 || len(resourceMetrics.Capacity) == 0 {
+		err := fmt.Errorf("error to aggregate container usage data using %s: used or capacity data points list is empty", maxUsageDataAggregator)
```

Comment:
Nit: as previous comment.
Suggestion (can be ignored here if you have more changes in pipeline): the common code within all aggregators can go into a common function.

Reply:
Good point. Thanks. Created common functions